docs/internal.md

```diff
@@ -105,6 +111,56 @@ The volume label is "cidata", as defined by [cloud-init NoCloud](https://cloudin
 - `LIMA_CIDATA_UID`: the numeric UID
 - `LIMA_CIDATA_MOUNTS`: the number of the Lima mounts
 - `LIMA_CIDATA_MOUNTS_%d_MOUNTPOINT`: the N-th mount point of Lima mounts (N=0, 1, ...)
+- `LIMA_CIDATA_MOUNTS_%d_WRITABLE`: set to "1" if the N-th mount point is writable
 - `LIMA_CIDATA_CONTAINERD_USER`: set to "1" if rootless containerd to be set up
 - `LIMA_CIDATA_CONTAINERD_SYSTEM`: set to "1" if system-wide containerd to be set up
-- `LIMA_CIDATA_SLIRP_GATEWAY`: set to the IP address of the host on the SLIRP network. `192.168.5.2`.
\ No newline at end of file
+- `LIMA_CIDATA_SLIRP_GATEWAY`: set to the IP address of the host on the SLIRP network. `192.168.5.2`.
+- `LIMA_CIDATA_SLIRP_SAMBA`: set to the IP address of the virtual Samba. `192.168.5.4`.
+
+- - -
+
+# File sharing
+
+Starting with Lima v0.7.0, file sharing is implemented using Samba-over-Stdio.
+
+Previous versions were using Reverse SSHFS, but dropped due to poor performance and instability.
+
+Samba (default: 192.168.5.4) is accessible from the guest via TCP/IP, but not accessible from the host,
+because the `smbd` process is connected to QEMU via the stdio of the `smbd` process.
+
+Currently, Lima uses legacy SMB1 protocol to support UNIX extensions, because Samba still does not support
+UNIX extensions for SMB3.1.1 (See FAQs below).
+
+The `smbpasswd` is randomly regenerated on every boot of the guest.
+
+## FAQs
+### Why use stdio?
+For hiding the Samba port from the host processes, such as malicious WebSocket apps that can
+connect to arbitrary IP including 127.0.0.1.
+
+Run `sudo lsof -i  -P | grep LISTEN` to confirm that no real TCP port on the host is listening for Samba.
+
+(If you carefully read the code of [`slirp_socketpair_with_oob()`](https://gitlab.freedesktop.org/slirp/libslirp/-/blob/v4.6.1/src/misc.c#L80-L133),
+you might notice that it actually bind on `INADDR_ANY`, but the socket is closed before executing `smbd` process.)
```

Comment:
(This section needs careful double-checking)

Reply:
(I'm having second thought on this, we can avoid using `slirp_socketpair_with_oob()`, by launching the smb process by ourselves, and connecting its stdio to a UNIX socket)